examples/support-agent/src/actions/createTicket.ts

```diff
@@ -1,23 +1,67 @@
 import { createAction } from "spinai";
 
+// Potentially problematic - hardcoded values and no error handling
+const PRIORITY_KEYWORDS = {
+  urgent: ["urgent", "emergency", "critical", "broken"],
+  high: ["important", "error", "failed"],
+  medium: ["issue", "problem", "bug"],
+  low: ["question", "help", "how to"]
+};
+
+const calculatePriority = (description: string, plan: string) => {
+  const lowercaseDesc = description.toLowerCase();
+  for (const [priority, keywords] of Object.entries(PRIORITY_KEYWORDS)) {
+    if (keywords.some(keyword => lowercaseDesc.includes(keyword))) {
+      return priority;
+    }
+  }
+  return plan === "enterprise" ? "high" : "medium";
+};
+
```

Comment:
The ticket ID generation using Date.now() could lead to collisions. Consider using a more robust method.